File: boj/BinarySearch/BOJ_14921.cpp
// 백준 14921 용액 합성하기
// 21.02.18
#include <iostream>
#include <algorithm>
#include <queue>
#include <vector>
#include <string>
#include <cmath>
#include <set>
#define INF 2147000000

typedef long long ll;

using namespace std;

int main()
{
    int n; scanf("%d", &n);
    vector <int> v;
    for (int i = 0; i < n; ++i)
    {
        int a; scanf("%d", &a);
        v.push_back(a);
    }

    int l = 0, r = n - 1, ans = INF;
    while (l < r)
    {
        int temp = v[l] + v[r];
        if (abs(ans) > abs(temp)) ans = temp;

        if (temp == 0) break;
        else if (temp < 0) ++l;
        else --r;
    }

    printf("%d\n", ans);

    return 0;
}

File: boj/BinarySearch/BOJ_14921.swift
//  백준 14921 용액 합성하기
//  21.02.18

import Foundation
let INF: Int = 2147000000

let n = Int(readLine()!)!
var arr = readLine()!.split(separator: " ").map { Int(String($0))! }

var l = 0, r = n - 1, ans = INF

while l < r {
  let temp = arr[l] + arr[r]
  if abs(ans) > abs(temp) { ans = temp }
  
  if temp == 0 { break }
  else if temp < 0 { l += 1 }
  else { r -= 1 }
}
print(ans)
